### PR TITLE
Added HTTP Authentication for Client

### PR DESCRIPTION
**Pull request by driftingaway86**
#535 
Attempting to add support for HTTP basic and beaer authentication for the http client. While this could be done manually, I felt it was a common enough thing to have to be their own functions respectively.

side note, first pull request ever.

### CODE REVIEW

**Review by DoumanAsh — APPROVED** [review]
Seems to be fine

**Comment by DoumanAsh** [comment]
Thanks for PR.
Could you please add mention of new methods to CHANGELOG?

**Comment by fafhrd91** [comment]
@f0cus3d could you also add test?

**Comment by driftingaway86** [comment]
> @f0cus3d could you also add test?

Yes, I was thinking about the test last night when the build verification was processing. I will get a test added sometime today and resubmit.

**Comment by fafhrd91** [comment]
test could as simple as just check if authorization header is set to write value

**Comment by jgrimes** [comment]
Hi, whats the status of this PR? Anything I can help with?

**Comment by crusty-dave** [comment]
I am not familiar with the process, when does this actually make it into nightly and then stable?
(I could actually use this now, although it is easy to code around.)

It looks like I would call it here:

```
pub fn query (request: &str) {
    client::ClientRequest::get(&request)   // <- Create request builder
        .header("User-Agent", "Actix-web")
>>        .base_auth("test", Some("test"))
        .finish().unwrap()
        .send()
        .conn_timeout(Duration::from_secs(10))// <- Send http request
        .map_err(|_| ())
        .and_then(handle_response ) .map_err(|_| ()
    ).poll().unwrap();
}

```

Thanks,
-Dave

**Comment by driftingaway86** [comment]
Hey all I added two tests on the client side for checking the add for basic and bearer auth. Apologies for the delay, have been travelling with the family.

File: tests/test_client.rs [hunk 1]
```diff
@@ -506,3 +506,31 @@ fn client_read_until_eof() {
...
 }
+
+#[test]
+fn client_basic_auth() {
+    let mut srv =
+        test::TestServer::new(|app| app.handler(|_| HttpResponse::Ok().body(STR)));
+
+    let request = srv
+                    .get()
+                    .basic_auth("username",Some("password"))
```
Comment (DoumanAsh) [inline tests/test_client.rs]: Space after `,`

**Review by DoumanAsh — APPROVED** [review]
Thank you for update, could you please apply simple comment and rebase your changes?

File: tests/test_client.rs (hunk 1 above)
Comment (driftingaway86) [reply tests/test_client.rs]: @DoumanAsh 

- Updated the spacing.
- Added comments above the requests in test_client.rs.
- rebase.

**Review by DoumanAsh — APPROVED** [review]
Thanks

**Comment by DoumanAsh** [comment]
@f0cus3d I've restarted failed travis job, once it is done I'll merge

**Comment by driftingaway86** [comment]
<3

**Comment by fafhrd91** [comment]
Just a note, if we want to keep this in 1.0, it should be ported to actix-http

**Comment by DoumanAsh** [comment]
@f0cus3d Are you up to porting it to [actix-http](https://github.com/actix/actix-http) ?

**Comment by driftingaway86** [comment]
@DoumanAsh 
Sure, adding it to my to do list for the day.

**Comment by driftingaway86** [comment]
@DoumanAsh @fafhrd91 
https://github.com/actix/actix-http/pull/11

**Pull request merged**